module/VuFind/src/VuFind/Sitemap/Plugin/ContentPages.php

```diff
@@ -0,0 +1,172 @@
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers Wiki
+ */
+namespace VuFind\Sitemap\Plugin;
+
+use Laminas\Config\Config;
+use Laminas\Router\RouteStackInterface;
+use VuFind\Sitemap\Sitemap;
+use VuFindTheme\ThemeInfo;
+
+/**
+ * Content pages generator plugin
+ *
+ * @category VuFind
+ * @package  Sitemap
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers Wiki
+ */
+class ContentPages extends AbstractGeneratorPlugin
+{
+    /**
+     * Theme informations
+     *
+     * @var ThemeInfo
+     */
+    protected $themeInfo;
+
+    /**
+     * Router
+     *
+     * @var RouteStackInterface
+     */
+    protected $router;
+
+    /**
+     * Base URL for site
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * Main VuFind configuration (config.ini)
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Patterns of files to be included
+     *
+     * @var array
+     */
+    protected $includedFiles = [
+        'templates/content/*.phtml',
+        'templates/content/*.md',
+    ];
+
+    /**
+     * Files to be ignored when searching for content pages
+     *
+     * @var array
+     */
+    protected $excludedFiles = [
+        'templates/content/content.phtml', // Content main page
+        'templates/content/markdown.phtml', // Content main page for Markdown
+    ];
+
+    /**
+     * Constructor
+     *
+     * @param ThemeInfo           $themeInfo Theme info
+     * @param RouteStackInterface $router    Router
+     * @param Config              $config    Main VuFind configuration
+     */
+    public function __construct(ThemeInfo $themeInfo, RouteStackInterface $router,
+        Config $config
+    ) {
+        $this->themeInfo = $themeInfo;
+        $this->router = $router;
+        $this->config = $config;
+    }
+
+    /**
+     * Set plugin options.
+     *
+     * @param array $options Options
+     *
+     * @return void
+     */
+    public function setOptions(array $options): void
+    {
+        parent::setOptions($options);
+        $this->baseUrl = $options['baseUrl'] ?? '';
+    }
+
+    /**
+     * Get the name of the sitemap used to create the sitemap file.
+     *
+     * @return string
+     */
+    public function getSitemapName(): string
+    {
+        return 'pages';
+    }
+
+    /**
+     * Add urls to the sitemap.
+     *
+     * @param Sitemap $sitemap Sitemap to add to
+     *
+     * @return void
+     */
+    public function addUrls(Sitemap $sitemap): void
+    {
+        $files = $this->themeInfo->findInThemes($this->includedFiles);
+        $nonLanguageFiles = [];
+        $languages = array_keys($this->config->Languages->toArray());
```

Comment:
Now that the slm/locale PR has been merged, it might be better to use LocaleSettings::getEnabledLocales() here instead of direct config access.

Reply:
Since $this->config isn't used for anything else, I'd suggest just taking in the language list through the constructor and putting the logic to fetch the list from LocaleSettings into the factory... that keeps the class itself simple and easier to test.

Reply:
Done.